system/security/securesocket/securesocket.cpp

```diff
@@ -244,7 +246,11 @@ class CSecureSocket : implements ISecureSocket, public CInterface
     //
     virtual bool set_nonblock(bool on) // returns old state
     {
-        throw MakeStringException(-1, "CSecureSocket::set_nonblock: not implemented");
+        bool prevState = m_socket->set_nonblock(on);
+        int flags = fcntl(SSL_get_fd(m_ssl), F_GETFL, 0);
```

Comment:
minor comment - why get the flags and confirm after setting ?  We set this in CSocket::set_nonblock() and that should be good enough ?

Reply:
true, I'll change.